src/libcollections/vec_deque.rs

```diff
@@ -1631,18 +1785,58 @@ impl<T> DoubleEndedIterator for IntoIter<T> {
 impl<T> ExactSizeIterator for IntoIter<T> {}
 
 /// A draining VecDeque iterator
-#[unstable(feature = "drain",
-           reason = "matches collection reform specification, waiting for dust to settle")]
+#[unstable(feature = "drain", reason = "recently added")]
 pub struct Drain<'a, T: 'a> {
-    inner: &'a mut VecDeque<T>,
+    after_tail: usize,
+    after_head: usize,
+    iter: IterMut<'a, T>,
```

Comment:
This doesn't have to be IterMut, it could just be an `Iter`. We only extract the elements, don't need to modify them. This is something I realized after vec's drain was done, so it's still Iter.